src/utilities/idf/IdfObject.cpp

```diff
@@ -825,14 +829,14 @@ namespace detail {
       StringVector temp = result;
       IdfExtensibleGroup eg = getExtensibleGroup(i);
       OS_ASSERT(!eg.empty());
-      result = eg.fields();
+      result = eg.fieldsWithHandles();
```

Comment:
Ok, so here is develop version of eraseExtensibleGroup: https://github.com/NREL/OpenStudio/blob/f916fc13ef0f90bf42b100e2050455f75c404667/src/utilities/idf/IdfObject.cpp#L806-L852


Basically it goes:

* popExtensibleGroup (remove last extensible group). This will actually resolve to WorkspaceObject_Impl::popExtensibleGroup which nullify the pointers too
* Then loop from last to groupIndex and copy groups / rollup until overwrite groupIndex
    * That rollup for each does basically this 
```
for i in [numExtensibleGroups() -1, groupIndex):
   prev_fields = extensibleGroups[i+1].fields() 
   extensibleGroups[i].setFields(prev_fields)
```
   
* The issue is that IdfExtensibleGroup::fields()  calls m_impl->getString which as reminder for a WorkspaceObject, actually **resolves** any objectList field to the Target workspaceobject's name, rather than return the handle.
 
* setFields in turns calls setString, which for a workspace object, if passed an object name for an objectList field, will convert to a target handle via `WorkspaceObject_Impl::convertToTargetHandle` which calls `WorkspaceObject_Impl::getObjectsByName`. If you have more than one candidate, it looks at reference lists too (but `OS:ZoneHVAC:EquipmentList` 's Equipment Name has `\reference AllObjects` (maybe that's a mistake too, but outside the point here), but bottom line is that if there are **more than one candidate, it returns the first**.

This means that in the case of a ZoneHVACEquipmentList that points to an Equipment named `XXXX` and you do have another modelObject named `XXXX` and it is found first, the call sequence goes:

```
prev_fields = ["XXXX", "", "", ""]
eg.setFields(prev_fields)
   eg.setString(0, "XXXX")
     handleOfFirstModelObjectNamedXXXX = "the handle of the wrong modelobject"
      eg.setString(0, handleOfFirstModelObjectNamedXXXX) => succeed because AllObjects allows it
```

Reply:
My fix is to always grab source fields as handles.

```
prev_fields = ["{handle of equipment}", "", "", ""]
eg.setFields(prev_fields)
      eg.setPointer(0, "{handle of equipment}")
```

Reply:
Is this also a peformance improvement? Surely yes. 

Thanks for the narrative Julien.